File: market/test/integration/java/ru/yandex/market/logistics/utilizer/config/DbQueueProducersConfig.java
package ru.yandex.market.logistics.utilizer.config;

import org.mockito.Mockito;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import ru.yandex.market.logistics.utilizer.dbqueue.task.events.SendUtilizationCycleFinalizationEmailProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.callticket.CallTicketDbqueueProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.closing.AutoCloseDbqueueProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.stockevent.SkuStocksEventDbqueueProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.stockevent.UnparsedSkuStocksFromLbDbqueueProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.transfer.CreateTransferDbqueueProducer;
import ru.yandex.market.logistics.utilizer.dbqueue.task.events.withdraw.WithdrawFileDbqueueProducer;

@Configuration
public class DbQueueProducersConfig {

    @Bean
    public SkuStocksEventDbqueueProducer skuStocksEventDbqueueProducer() {
        return Mockito.mock(SkuStocksEventDbqueueProducer.class);
    }

    @Bean
    public UnparsedSkuStocksFromLbDbqueueProducer unparsedSkuStocksFromLbDbqueueProducer() {
        return Mockito.mock(UnparsedSkuStocksFromLbDbqueueProducer.class);
    }

    @Bean
    public CreateTransferDbqueueProducer createTransferDbqueueProducer() {
        return Mockito.mock(CreateTransferDbqueueProducer.class);
    }

    @Bean
    public SendUtilizationCycleFinalizationEmailProducer sendUtilizationCycleFinalizationEmailProducer() {
        return Mockito.mock(SendUtilizationCycleFinalizationEmailProducer.class);
    }

    @Bean
    public CallTicketDbqueueProducer callTicketDbqueueProducer() {
        return Mockito.mock(CallTicketDbqueueProducer.class);
    }

    @Bean
    public WithdrawFileDbqueueProducer createWithdrawFileDbqueueProducer() {
        return Mockito.mock(WithdrawFileDbqueueProducer.class);
    }

    @Bean
    public AutoCloseDbqueueProducer autoCloseDbqueueProducer() {
        return Mockito.mock(AutoCloseDbqueueProducer.class);
    }
}

File: market/test/integration/kotlin/config/DbQueueProducersConfig.kt
package ru.yandex.market.logistics.calendaring.config

import org.mockito.Mockito
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.boot.test.mock.mockito.MockReset.after
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import ru.yandex.market.logistics.calendaring.dbqueue.producer.ProcessMetaInfoChangeEventProducer
import ru.yandex.market.logistics.calendaring.dbqueue.producer.ReleaseQuotaProducer

@Configuration
open class DbQueueProducersConfig {

    @ConditionalOnMissingBean(ReleaseQuotaProducer::class)
    @Bean
    open fun releaseQuotaProducer(): ReleaseQuotaProducer {
        return Mockito.mock(ReleaseQuotaProducer::class.java, after())
    }

    @ConditionalOnMissingBean(ProcessMetaInfoChangeEventProducer::class)
    @Bean
    open fun processMetaInfoChangeEventProducer(): ProcessMetaInfoChangeEventProducer {
        return Mockito.mock(ProcessMetaInfoChangeEventProducer::class.java)
    }
}
